src/Orleans.Reminders/ReminderService/ReminderRegistry.cs

```diff
@@ -48,6 +48,31 @@ public Task<IGrainReminder> RegisterOrUpdateReminder(GrainId callingGrainId, str
             return GetGrainService(callingGrainId).RegisterOrUpdateReminder(callingGrainId, reminderName, dueTime, period);
         }
 
+        public Task<bool> TryRegisterReminder(GrainId callingGrainId, string reminderName, TimeSpan dueTime, TimeSpan period)
+        {
+            // Perform input volatility checks that are consistent with System.Threading.Timer
+            // http://referencesource.microsoft.com/#mscorlib/system/threading/timer.cs,c454f2afe745d4d3,references
+            if (dueTime.Ticks < 0 && dueTime != Timeout.InfiniteTimeSpan)
+                throw new ArgumentOutOfRangeException(nameof(dueTime), "Cannot use negative dueTime to create a reminder");
+            if (dueTime.Ticks > MaxSupportedTimeout * TimeSpan.TicksPerMillisecond)
+                throw new ArgumentOutOfRangeException(nameof(dueTime), $"Cannot use value larger than {MaxSupportedTimeout}ms for dueTime when creating a reminder");
+
+            if (period.Ticks < 0 && period != Timeout.InfiniteTimeSpan)
+                throw new ArgumentOutOfRangeException(nameof(period), "Cannot use negative period to create a reminder");
+            if (period.Ticks > MaxSupportedTimeout * TimeSpan.TicksPerMillisecond)
+                throw new ArgumentOutOfRangeException(nameof(period), $"Cannot use value larger than {MaxSupportedTimeout}ms for period when creating a reminder");
+
+            var minReminderPeriod = options.MinimumReminderPeriod;
+            if (period < minReminderPeriod)
+                throw new ArgumentException($"Cannot register reminder {reminderName} as requested period ({period}) is less than minimum allowed reminder period ({minReminderPeriod})");
+
+            if (string.IsNullOrEmpty(reminderName))
+                throw new ArgumentException("Cannot use null or empty name for the reminder", nameof(reminderName));
```

Comment:
This should be moved into a separate ValidateParameters type of method so that the validation logic cannot diverge

Reply:
Can this suffice the need:
```csharp
    public Task<IGrainReminder> RegisterOrUpdateReminder(GrainId callingGrainId, string reminderName, TimeSpan dueTime, TimeSpan period)
    {
        ValidateParameters(callingGrainId, reminderName, dueTime, period);
        EnsureReminderServiceRegistered();
        return GetGrainService(callingGrainId).RegisterOrUpdateReminder(callingGrainId, reminderName, dueTime, period);
    }
    
    public Task<bool> TryRegisterReminder(GrainId callingGrainId, string reminderName, TimeSpan dueTime, TimeSpan period)
    {
        ValidateParameters(callingGrainId, reminderName, dueTime, period);
        EnsureReminderServiceRegistered();
        return GetGrainService(callingGrainId).TryRegisterReminder(callingGrainId, reminderName, dueTime, period);
    }
    
    private void ValidateParameters(GrainId callingGrainId, string reminderName, TimeSpan dueTime, TimeSpan period)
    {
        // Perform input volatility checks that are consistent with System.Threading.Timer
        // http://referencesource.microsoft.com/#mscorlib/system/threading/timer.cs,c454f2afe745d4d3,references
        if (dueTime.Ticks < 0 && dueTime != Timeout.InfiniteTimeSpan)
            throw new ArgumentOutOfRangeException(nameof(dueTime), "Cannot use negative dueTime to create a reminder");
        if (dueTime.Ticks > MaxSupportedTimeout * TimeSpan.TicksPerMillisecond)
            throw new ArgumentOutOfRangeException(nameof(dueTime), $"Cannot use value larger than {MaxSupportedTimeout}ms for dueTime when creating a reminder");
        if (period.Ticks < 0 && period != Timeout.InfiniteTimeSpan)
            throw new ArgumentOutOfRangeException(nameof(period), "Cannot use negative period to create a reminder");
        if (period.Ticks > MaxSupportedTimeout * TimeSpan.TicksPerMillisecond)
            throw new ArgumentOutOfRangeException(nameof(period), $"Cannot use value larger than {MaxSupportedTimeout}ms for period when creating a reminder");
        var minReminderPeriod = options.MinimumReminderPeriod;
        if (period < minReminderPeriod)
            throw new ArgumentException($"Cannot register reminder {reminderName} as requested period ({period}) is less than minimum allowed reminder period ({minReminderPeriod})");
        if (string.IsNullOrEmpty(reminderName))
            throw new ArgumentException("Cannot use null or empty name for the reminder", nameof(reminderName));
    }

```

Reply:
resolved using: fed10fe